doc/tools/coqrst/notations/sphinx.py

```diff
@@ -45,7 +45,11 @@ def is_alternative(node):
     def visitRepeat(self, ctx:TacticNotationsParser.RepeatContext):
         # Uses inline nodes instead of subscript and superscript to ensure that
         # we get the right customization hooks at the LaTeX level
-        wrapper = nodes.inline('', '', classes=['repeat-wrapper'])
+        separator = ctx.ATOM() or ctx.PIPE()
+        # todo: I wanted to have 2 independent classes "repeat-wrapper" and "with-sub" here,
+        # but that breaks the latex build (invalid .tex file)
```

Comment:
Surprising.  Normally this should get you two nested classes.  Can you show what it generated? Otherwise it's not a big issue the way you did it.

Reply:
With this change:

```
        wrapper = nodes.inline('', '', classes=[classes, 'bonus-class'])
```

The generated tex includes `\DUrole{repeat-wrapper,bonus-class}`

The latex console shows (where I typed "R"):

```
! Missing $ inserted.
<inserted text>
                $
l.2668 ...ern}}}}}\DUrole{notation-sup}{+}} {]}}}}
                                                   as^^M
? R
OK, entering \nonstopmode...
```

Line 2668 in the tex file is:

```
Syntax \sphinxcode{\sphinxupquote{\DUrole{notation}{injection {\hyperref[\detokenize{language/gallina-specification-language:grammar-token-term}]{\sphinxcrossref{\DUrole{hole}{term}}}} as {[}= \DUrole{repeat-wrapper,bonus-class}{\DUrole{repeat}{{\hyperref[\detokenize{proof-engine/tactics:grammar-token-intropattern}]{\sphinxcrossref{\DUrole{hole}{intropattern}}}}}\DUrole{notation-sup}{+}} {]}}}} as
```

Looks like Sphinx is not entirely consistent in how it processes `classes`.  Maybe not handled at a single point in the code.

Reply:
I see.  The trick I think would be to define the "joint" role `\DUrole{repeat-wrapper,bonus-class}` on the LaTeX side.

Reply:
A workaround.  Though it doesn't scale well for the general case: if there are N classes and all combinations are allowed, 2^N definitions would be needed.  I think what I have is OK.

Reply:
Yes, what we have is the workaround in both the latex and the html, right? I think that's fine.

Reply:
Yes, it works for both but it's a no-op (and not needed) in Latex.